src/NuGet.Core/NuGet.ProjectModel/CacheFile.cs

```diff
@@ -51,9 +59,8 @@ public override int GetHashCode()
             var combiner = new HashCodeCombiner();
             combiner.AddObject(DgSpecHash);
             combiner.AddObject(Version);
+            combiner.AddObject(ProjectFullPath);
```

Comment:
I'm using it to store the list of files which is written to disk.  When reading the file, I only look at the existence of each path and update the `AnyPackagesMissing` property and don't populate the `ExpectedFilePaths` array.  Since there's no guarantee that the property is filled, I left it out of the hashcode